src/lua/datetime.lua

```diff
@@ -864,6 +879,86 @@ local function datetime_interval_add(lhs, rhs)
     end
 end
 
+--[[
+    Parse partial ISO-8601 date string
+
+    Accepted formats are:
+
+    Basic      Extended
+    20121224   2012-12-24   Calendar date   (ISO 8601)
+    2012359    2012-359     Ordinal date    (ISO 8601)
+    2012W521   2012-W52-1   Week date       (ISO 8601)
+    2012Q485   2012-Q4-85   Quarter date
+
+    Returns pair of constructed datetime object, and length of string
+    which has been accepted by parser.
+]]
+local function datetime_parse_date(str)
+    check_str(str, "datetime.parse_date()")
+    local dt = ffi.new('dt_t[1]')
+    local len = tonumber(builtin.tnt_dt_parse_iso_date(str, #str, dt))
+    if len == 0 then
+        error(('invalid date format %s'):format(str), 2)
+    end
+    return datetime_new_dt(dt[0]), len
+end
+
+--[[
+    datetime parse function for strings in extended iso-8601 format
+    assumes to deal with date T time time_zone at once
+       date [T] time [ ] time_zone
+    Returns constructed datetime object and length of accepted string.
+]]
+local function datetime_parse_full(str, tzoffset)
+    check_str(str, "datetime.parse()")
+    local date = ffi.new(datetime_t)
+    local len = builtin.tnt_datetime_parse_full(date, str, #str, tzoffset)
+    if len == 0 then
+        error(("could not parse '%s'"):format(str))
+    end
+    return date, len
+end
+
+--[[
+    Parse datetime string given `strptime` like format.
+    Returns constructed datetime object and length of accepted string.
+]]
+local function datetime_parse_format(str, fmt)
+    local date = ffi.new(datetime_t)
+    local len = builtin.tnt_datetime_strptime(date, str, fmt)
+    if len == 0 then
+        error(("could not parse '%s' using '%s' format"):format(str, fmt))
+    end
+    return date, len
+end
+
+local function datetime_parse_from(str, obj)
+    check_str(str, "datetime.parse()")
+    local fmt = ''
+    local offset
+
+    if obj ~= nil then
+        check_table(obj, "datetime.parse()")
+        fmt = obj.format
+        offset = obj.tzoffset
+    end
+    check_str_or_nil(fmt, "datetime.parse()")
+
+    if offset ~= nil then
+        offset = get_timezone(offset, 'tzoffset')
+        check_range(offset, -720, 840, 'tzoffset')
+    end
+    if obj and obj.tz ~= nil then
+        nyi('tz')
```

Comment:
Mixed single quotes and double quotes in a whole source file.

Reply:
In majority of a cases I could not fix it (and don't see much reason why should I care at this stage), but for the newly introduced `datetime_parse*` functions I've converted some double-quotes to single-quotes, except a couple of cases where we need to output single-quotes.

```diff
commit f71ec7e0b303c912c7b99006c4cc1246dafbe210
Author: Timur Safin <tsafin@tarantool.org>
Date:   Fri Jan 28 17:01:38 2022 +0300

    temp - make quotes more consistent in functions

diff --git a/src/lua/datetime.lua b/src/lua/datetime.lua
index e74313029..496e15772 100644
--- a/src/lua/datetime.lua
+++ b/src/lua/datetime.lua
@@ -894,7 +894,7 @@ end
     which has been accepted by parser.
 ]]
 local function datetime_parse_date(str)
-    check_str(str, "datetime.parse_date()")
+    check_str(str, 'datetime.parse_date()')
     local dt = ffi.new('dt_t[1]')
     local len = tonumber(builtin.tnt_dt_parse_iso_date(str, #str, dt))
     if len == 0 then
@@ -910,7 +910,7 @@ end
     Returns constructed datetime object and length of accepted string.
 ]]
 local function datetime_parse_full(str, tzoffset)
-    check_str(str, "datetime.parse()")
+    check_str(str, 'datetime.parse()')
     local date = ffi.new(datetime_t)
     local len = builtin.tnt_datetime_parse_full(date, str, #str, tzoffset)
     if len == 0 then
@@ -933,16 +933,16 @@ local function datetime_parse_format(str, fmt)
 end
 
 local function datetime_parse_from(str, obj)
-    check_str(str, "datetime.parse()")
+    check_str(str, 'datetime.parse()')
     local fmt = ''
     local offset
 
     if obj ~= nil then
-        check_table(obj, "datetime.parse()")
+        check_table(obj, 'datetime.parse()')
         fmt = obj.format
         offset = obj.tzoffset
     end
-    check_str_or_nil(fmt, "datetime.parse()")
+    check_str_or_nil(fmt, 'datetime.parse()')
 
     if offset ~= nil then
         offset = get_timezone(offset, 'tzoffset')
```

Reply:
Thanks!